infra/aws/terraform/prow-build-cluster/eks.tf

```diff
@@ -133,3 +163,29 @@ module "eks" {
     }
   }
 }
+
+# We create IAM role to delegate access to the cluster.
+resource "aws_iam_role" "iam_cluster_admin" {
+  name        = "Prow-Cluster-Admin"
+  description = "IAM role used to delegate access to prow-build-cluster"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        "Effect" : "Allow",
+        "Principal" : {
+          "AWS" : "468814281478"
```

Comment:
It would be nice to have `468814281478` in a named variable.

Reply:
we have `root_account_arn` defined below as a variable but I would prefer we use the data resource [caller_identity](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/caller_identity).

Reply:
If we want to go that route, I think we should create IAM user group and give access to it. Please note that this is currently supposed to be used by users manually (e.g. me and @pkprzekwas), eventually by some automation (but for automation I would go via IRSA).

Reply:
I changed this so only @pkprzekwas and I can assume this role.